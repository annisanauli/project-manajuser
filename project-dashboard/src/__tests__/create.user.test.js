import React from 'react';
import CreateUser, {Role} from '../models/create.user';
import Enzyme, {shallow, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({adapter:new Adapter()});

describe("create.user component", () => {
  test("renders", () => {
    const wrapper = shallow(<CreateUser />);
    //expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should capture name correctly onChange', () =>{
    const component = mount(<CreateUser />);
    const input = component.find('input').at(0);
    input.instance().value = 'Nauli';
    input.simulate('change');
    expect(component.state().fields["campName"]).toEqual('Nauli');
    expect(component).toMatchSnapshot();
  })

  it('Should capture email correctly onChange', () =>{
    const component = mount(<CreateUser />);
    const input = component.find('input').at(1);
    input.instance().value = 'nauli@gmail.com';
    input.simulate('change');
    expect(component.find('input').at(1).props().value).toEqual('nauli@gmail.com');
    //expect(component.state().fields["campEmail"]).toEqual('nauli@gmail.com');
    expect(component).toMatchSnapshot();
   })

   it('Should capture password correctly onChange', () =>{
    const component = mount(<CreateUser />);
    const input = component.find('input').at(2);
    input.instance().value = 'Nauli17_';
    input.simulate('change');
    //expect(component.state().fields["campPassword"]).toEqual('Nauli17_');
    expect(component.find('input').at(2).props().value).toEqual('Nauli17_');
    expect(component).toMatchSnapshot();
  })

  it('Should capture phone correctly onChange', () =>{
    const component = mount(<CreateUser />);
    const input = component.find('input').at(3);
    input.instance().value = '08126312810';
    input.simulate('change');
    //expect(component.state().fields["campPhone"]).toEqual('08126312810');
    expect(component.find('input').at(3).props().value).toEqual('08126312810');
    expect(component).toMatchSnapshot();
  })

  it('Should capture select role correctly onChange', () =>{
    const component = mount(<CreateUser />);
    const input = component.find('select').at(0);
    const optionRole = component.find('option').at(1);
    optionRole.instance().selected  = true;
    input.simulate('change');
    //expect(component.find('select').at(0).props().value).toEqual('Role 2');
    expect(component.state().selectRole).toEqual('1');
    expect(component).toMatchSnapshot();
  });

  it('calls handleSubmit prop function when form is submitted', () => {
    const wrapper = shallow(<CreateUser />);
    const fn = jest.fn();
    wrapper.instance().handleSubmit = fn;
    wrapper.instance().handleSubmit();
        wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
      
    });
    //wrapper.setState(state)
    expect(fn).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
  
})