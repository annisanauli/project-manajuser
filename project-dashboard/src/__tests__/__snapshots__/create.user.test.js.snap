// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nyoba testing project-dasboard 1`] = `
<div>
  <div
    className="box-body"
  >
    <form
      className="contactform"
      name="contactform"
      onSubmit={[Function]}
    >
      <fieldset>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputName1"
          >
            Nama
          </label>
          <input
            Required={true}
            class="form-control"
            onChange={[Function]}
            placeholder="Name"
            size="30"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputHandphone1"
          >
            No. Handphone
          </label>
          <input
            class="form-control"
            onChange={[Function]}
            placeholder="Phone"
            refs="phone"
            size="12"
            type="telp"
          />
          <span
            class="help-block"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Email
          </label>
          <input
            class="form-control"
            onChange={[Function]}
            placeholder="Email"
            refs="email"
            size="30"
            type="email"
          />
          <span
            class="help-block"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputPassword1"
          >
            Password
          </label>
          <input
            class="form-control"
            onChange={[Function]}
            placeholder="Password"
            refs="password"
            size="15"
            type="password"
          />
          <span
            class="help-block"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="inputState"
          >
            User Role
          </label>
          <select
            class="form-control"
            id="inputState"
            onChange={[Function]}
          >
            <option
              selected={true}
            >
              Choose...
            </option>
            <option
              value="1"
            >
              Role 1
            </option>
            <option
              value="2"
            >
              Role 2
            </option>
            <option
              value="3"
            >
              Role 3
            </option>
          </select>
        </div>
      </fieldset>
      <div
        className="col-md-12"
      >
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-default"
            data-dismiss="modal"
            type="button"
          >
            Batal
          </button>
          <button
            class="btn btn-primary"
            id="submit"
            type="submit"
          >
            Simpan
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
